src/main/java/spoon/reflect/CtModelImpl.java

```diff
@@ -39,6 +43,23 @@
 
 	private static final long serialVersionUID = 1L;
 
+	@Override
+	public <R extends CtElement> CtQuery filterChildren(Filter<R> filter) {
+		return getUnnamedModule().getFactory().Query().createQuery(this.getAllModules().toArray()).filterChildren(filter);
```

Comment:
`QueryFactory#createQuery` has actually parameter type Object, not Object[], so this code will not work as expected. We have to add new QueryFactory#createQuery(Object[]) method.

Reply:
My bad, that's because `CtQueryImpl` constructor as an `Object...` arg. I'll fix that.